java/arcs/core/host/Particle.kt

```diff
@@ -35,4 +44,7 @@ interface Particle {
      * @param allSynced flag indicating if all handles are synchronized
      */
     suspend fun onHandleSync(handle: Handle, allSynced: Boolean) = Unit
+
+    /** Called when an [Arc] is shutdown. */
```

Comment:
It would be good to mention examples of things that the particle author might find useful to perform during `onShutdown`